unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -423,5 +438,21 @@ private void AddFriendRequests(AddFriendRequestsPayload msg)
                     { action = FriendshipAction.REQUESTED_TO, userId = userId });
             }
         }
+
+        private FriendshipAction ToFriendshipAction(FriendshipStatus status)
+        {
+            switch (status)
+            {
+                case FriendshipStatus.FRIEND:
+                    return FriendshipAction.APPROVED;
+                case FriendshipStatus.NOT_FRIEND:
+                default:
+                    return FriendshipAction.NONE;
+                case FriendshipStatus.REQUESTED_TO:
+                    return FriendshipAction.REQUESTED_TO;
+                case FriendshipStatus.REQUESTED_FROM:
+                    return FriendshipAction.REQUESTED_FROM;
+            }
+        }
```

Comment:
this Singleton is already grew up to 456 lines of code. Could be a good to split it to with respect to different responsibilities and then compose it from these smaller focused classes